cpp/include/cuspatial/detail/trajectory.hpp

```diff
@@ -47,14 +51,31 @@ std::unique_ptr<cudf::experimental::table> trajectory_distances_and_speeds(
 
 /**
  * @copydoc cudf::experimental::trajectory_bounding_boxes()
- * @param[in] stream Optional CUDA stream on which to schedule allocations
+ * @param stream Optional CUDA stream on which to schedule allocations
  */
 std::unique_ptr<cudf::experimental::table> trajectory_bounding_boxes(
     cudf::column_view const& object_id, cudf::column_view const& x,
     cudf::column_view const& y,
     rmm::mr::device_memory_resource* mr = rmm::mr::get_default_resource(),
     cudaStream_t stream = 0);
 
+/**
+ * @brief Count the number of unique object ids
+ *
+ * @param object_id Column of object (e.g., vehicle) ids
+ * @param stream Optional CUDA stream on which to schedule allocations
+ * @return cudf::size_type The number of unique elements
+ */
+inline cudf::size_type count_unique_ids(cudf::column_view const& object_id,
+                                        cudaStream_t stream = 0) {
+  auto policy = rmm::exec_policy(stream);
+  rmm::device_vector<int32_t> unique_keys(object_id.size());
```

Comment:
Now that I see this, if the purpose is only to count unique ids, then this is quite inefficient in memory overhead and bandwidth. Instead, I would do a transform_reduce with one of the input iterators offset by one. The transform just returns 1 if the values are different, 0 if they are the same. The reduction operator is just sum.

Reply:
If you prefer, we can file an issue to fix this in a separate PR.